src/Datadog.Trace/Sampling/RegexSamplingRule.cs

```diff
@@ -0,0 +1,143 @@
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Sampling
+{
+    internal class RegexSamplingRule : ISamplingRule
+    {
+        private static readonly Vendors.Serilog.ILogger Log = DatadogLogging.For<RegexSamplingRule>();
+
+        private readonly float _samplingRate;
+        private readonly string _serviceNameRegex;
+        private readonly string _operationNameRegex;
+
+        public RegexSamplingRule(
+            float rate,
+            string name,
+            string serviceNameRegex,
+            string operationNameRegex)
+        {
+            _samplingRate = rate;
+            _serviceNameRegex = serviceNameRegex;
+            _operationNameRegex = operationNameRegex;
+
+            Name = name;
+        }
+
+        public string Name { get; }
+
+        /// <summary>
+        /// Gets the Priority of the rule.
+        /// Configuration rules will default to 1 as a priority and rely on order of specification.
+        /// </summary>
+        public int Priority => 1;
+
+        public static IEnumerable<RegexSamplingRule> BuildFromConfigurationString(string configuration)
+        {
+            if (!string.IsNullOrEmpty(configuration))
+            {
+                var ruleStrings = configuration.Split(new[] { ";", ":" }, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
Splitting on both `;` and `:` was a nice way of solving this for different OSes.